### ISSUE TITLE
Gpipe: Efficient training of giant neural networks using pipeline parallelism

### ISSUE THREAD

**Issue by sypark9646** [sypark9646/paper-logs#7]
## 어떤 내용의 논문인가요? 👋

대형 딥 뉴럴 네트워크 모델을 효율적으로 모델 병렬화 하는 방법에 관한 연구

## Abstract (요약) 🕵🏻‍♂️

Scaling up deep neural network capacity has been known as an effective approach to improving model quality for several different machine learning tasks. In many cases, increasing model capacity beyond the memory limit of a single accelerator has required developing special algorithms or infrastructure. These solutions are often architecture-specific and do not transfer to other tasks. To address the need for efficient and task-independent model parallelism, we introduce GPipe, a pipeline parallelism library that allows scaling any network that can be expressed as a sequence of layers. By pipelining different sub-sequences of layers on separate accelerators, GPipe provides the flexibility of scaling a variety of different networks to gigantic sizes efficiently. Moreover, GPipe utilizes a novel batch-splitting pipelining algorithm, resulting in almost linear speedup when a model is partitioned across multiple accelerators. We demonstrate the advantages of GPipe by training large-scale neural networks on two different tasks with distinct network architectures: (i) Image Classification: We train a 557-million-parameter AmoebaNet model and attain a top-1 accuracy of 84.4% on ImageNet-2012, (ii) Multilingual Neural Machine Translation: We train a single 6-billion-parameter, 128-layer Transformer model on a corpus spanning over 100 languages and achieve better quality than all bilingual models.

## 이 논문을 읽어서 무엇을 배울 수 있는지 알려주세요! 🤔

모델 병렬화, 파이프라이닝, Rematerialization

## 같이 읽어보면 좋을 만한 글이나 이슈가 있을까요?

https://github.com/kakaobrain/torchgpipe

**Comment by sypark9646** [comment]
https://sysgongbu.tistory.com/214
